enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/DefaultAnalyticsEventPublisher.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.enforcer.analytics;
+
+import io.envoyproxy.envoy.data.accesslog.v3.HTTPAccessLogEntry;
+import io.envoyproxy.envoy.service.accesslog.v3.StreamAccessLogsMessage;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.carbon.apimgt.common.analytics.AnalyticsCommonConfiguration;
+import org.wso2.carbon.apimgt.common.analytics.AnalyticsServiceReferenceHolder;
+import org.wso2.carbon.apimgt.common.analytics.collectors.AnalyticsDataProvider;
+import org.wso2.carbon.apimgt.common.analytics.collectors.impl.GenericRequestDataCollector;
+import org.wso2.carbon.apimgt.common.analytics.exceptions.AnalyticsException;
+import org.wso2.carbon.apimgt.common.analytics.publishers.dto.enums.EventCategory;
+import org.wso2.carbon.apimgt.common.analytics.publishers.dto.enums.FaultCategory;
+import org.wso2.choreo.connect.enforcer.constants.AnalyticsConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Default Analytics Event publisher to the analytics cloud.
+ */
+public class DefaultAnalyticsEventPublisher implements AnalyticsEventPublisher {
+    private static final String AUTH_TOKEN_KEY = "auth.api.token";
+    private static final String AUTH_URL = "auth.api.url";
+    private static boolean isChoreoDeployment = false;
+
+    private static final Logger logger = LogManager.getLogger(DefaultAnalyticsEventPublisher.class);
+
+    @Override
+    public void handleGRPCLogMsg(StreamAccessLogsMessage message) {
+        for (int i = 0; i < message.getHttpLogs().getLogEntryCount(); i++) {
+            HTTPAccessLogEntry logEntry = message.getHttpLogs().getLogEntry(i);
+            logger.trace("Received logEntry from Router " + message.getIdentifier().getNode() +
+                    " : " + message.toString());
+            if (doNotPublishEvent(logEntry)) {
+                logger.debug("LogEntry is ignored as it is already published by the enforcer.");
+                continue;
+            }
+            AnalyticsDataProvider provider = new MgwAnalyticsProvider(logEntry, isChoreoDeployment);
```

Comment:
Rename the class name

Reply:
fixed via https://github.com/wso2/product-microgateway/pull/1883/commits/e8f0a4dbd3322e7a782b875fc3b09238013ef33f